From 40397851ad02fc5eb970a665b5e68075da824097 Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Mon, 11 Nov 2024 10:54:20 +0100
Subject: [PATCH] respect BUILD_SHARED_LIBS/BUILD_STATIC_LIBS in FORTRAN

---
 FORTRAN/CMakeLists.txt | 75 ++++++++++++++++++++++++++----------------
 1 file changed, 47 insertions(+), 28 deletions(-)

diff --git a/FORTRAN/CMakeLists.txt b/FORTRAN/CMakeLists.txt
index 2efa03dd..472f57d7 100644
--- a/FORTRAN/CMakeLists.txt
+++ b/FORTRAN/CMakeLists.txt
@@ -15,37 +15,52 @@ if(enable_complex16)
   list(APPEND sources c2f_zcreate_matrix_x_b.c superlu_c2f_zwrap.c)
 endif()  
 
-add_library(superlu_dist_fortran ${sources})
-add_library(superlu_dist_fortran-static STATIC ${sources})
+if(BUILD_SHARED_LIBS)
+  add_library(superlu_dist_fortran ${sources})
+endif()
+if(BUILD_STATIC_LIBS)
+  add_library(superlu_dist_fortran-static STATIC ${sources})
+endif()
 # set(targets superlu_dist_fortran)
 get_target_property(superlu_dist_version superlu_dist VERSION)
 get_target_property(superlu_dist_soversion superlu_dist SOVERSION)
-set_target_properties(superlu_dist_fortran PROPERTIES VERSION ${superlu_dist_version})
-set_target_properties(superlu_dist_fortran PROPERTIES SOVERSION ${superlu_dist_soversion})
-target_link_libraries(superlu_dist_fortran superlu_dist)
-
-set_target_properties(superlu_dist_fortran-static PROPERTIES OUTPUT_NAME superlu_dist_fortran)
-set_target_properties(superlu_dist_fortran-static PROPERTIES VERSION ${superlu_dist_version})
-set_target_properties(superlu_dist_fortran-static PROPERTIES SOVERSION ${superlu_dist_soversion})
-target_link_libraries(superlu_dist_fortran-static superlu_dist)
+if(BUILD_SHARED_LIBS)
+  set_target_properties(superlu_dist_fortran PROPERTIES VERSION ${superlu_dist_version})
+  set_target_properties(superlu_dist_fortran PROPERTIES SOVERSION ${superlu_dist_soversion})
+  target_link_libraries(superlu_dist_fortran superlu_dist)
+endif()
+if (BUILD_STATIC_LIBS)
+  set_target_properties(superlu_dist_fortran-static PROPERTIES OUTPUT_NAME superlu_dist_fortran)
+  set_target_properties(superlu_dist_fortran-static PROPERTIES VERSION ${superlu_dist_version})
+  set_target_properties(superlu_dist_fortran-static PROPERTIES SOVERSION ${superlu_dist_soversion})
+  target_link_libraries(superlu_dist_fortran-static superlu_dist)
+endif()
 
 # depends on FPP defs
-add_dependencies(superlu_dist_fortran config_f)
-add_dependencies(superlu_dist_fortran-static config_f)
-add_dependencies(superlu_dist_fortran-static superlu_dist_fortran)
-
-install(TARGETS superlu_dist_fortran
-# DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
-    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
-)
-install(TARGETS superlu_dist_fortran-static
-# DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
-    LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
-    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
-)
+if(BUILD_SHARED_LIBS)
+  add_dependencies(superlu_dist_fortran config_f)
+endif()
+if(BUILD_STATIC_LIBS)
+  add_dependencies(superlu_dist_fortran-static config_f)
+  add_dependencies(superlu_dist_fortran-static superlu_dist_fortran)
+endif()
+
+if(BUILD_SHARED_LIBS)
+  install(TARGETS superlu_dist_fortran
+  # DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
+  )
+endif()
+if (BUILD_STATIC_LIBS)
+  install(TARGETS superlu_dist_fortran-static
+  # DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
+      LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
+      ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
+  )
+endif()
 
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
@@ -61,8 +76,12 @@ include_directories(${MPI_Fortran_INCLUDE_PATH})
 link_directories(${MPI_Fortran_LIBRARIES})
 
 # Libs to be linked with the Fortran codes
-set(fortran_link_libs superlu_dist_fortran ${MPI_Fortran_LIBRARIES} ${BLAS_LIB} ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
-set(fortran_link_libs superlu_dist_fortran-static ${MPI_Fortran_LIBRARIES} ${BLAS_LIB} ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
+if(BUILD_SHARED_LIBS)
+  set(fortran_link_libs superlu_dist_fortran ${MPI_Fortran_LIBRARIES} ${BLAS_LIB} ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
+endif()
+if(BUILD_STATIC_LIBS)
+  set(fortran_link_libs superlu_dist_fortran-static ${MPI_Fortran_LIBRARIES} ${BLAS_LIB} ${CMAKE_Fortran_IMPLICIT_LINK_LIBRARIES})
+endif()
 set(all_link_libs ${fortran_link_libs} superlu_dist)
 
 #message("!!! in Fortran: MPI_Fortran_LINK_FLAGS='${MPI_Fortran_LINK_FLAGS}'")
-- 
2.45.0

